Update to unified API, armv7s, arm64 and latest TTTAttributedLabel

beeradmoore opened the pull request:
TTTAttributedLabel updates
- Updated TTTAttributedLabel to the latest .m and .h files.
- Added build support for armv7s.
- Added build  support for arm64.

MonoTouch.TTTAttributedLabel updates
- Updated MonoTouch.TTTAttributedLabel to unified API
- Updated project to support armv7s and arm64.
- Added support for the following static values (TTTBackgroundCornerRadiusAttributeName, TTTBackgroundFillColorAttributeName, TTTBackgroundFillPaddingAttributeName, TTTBackgroundLineWidthAttributeName, TTTBackgroundStrokeColorAttributeName, TTTStrikeOutAttributeName)

Commit:
Updated to unified API.
Updated to include armv7s and arm64.

Commit:
Updated to latest TTTAttributedLabel.

beeradmoore commented:
"latest TTTAttributedLabel" appears to be from when I last got their code from github about a month ago.

lipka commented:
Thanks guys, will merge this in later today! Awesome work with updating to the latest version of TTTAttributedLabel.

Commit:
Added sln file.
Fixed binding issues with the 6 __Internal properties.
Updated TTTAttributedLablel to the latest version and added bindings to new functions.
Added NSTextCheckingResult and NSTextCheckingType from MonoMac.
Added LinkTarget to Simulator64
Restored deprecated functions but added the [Obsolete] tag.
Removed binding of Text to allow use of the SetText function due to a bug listed below. Can't have both of them because Text { get; set; } and binding to the setText: function causes a conflict. Unsure if the Text set; will call TTTAttributedLabels internal setText: method or not.

 @bug Setting `attributedText` directly is not recommended, as it may cause a crash when attempting to access any links previously set. Instead, call `setText:`, passing an `NSAttributedString`.

beeradmoore commented:
Added more to this today. Mainly got NSTextCheckingResult in.

As noted in my commit I removed the binding of Text. Text { get; set; } would bind somehow to the method setText: which would mean we couldn't bind SetText( to setText:. I am unsure if the set part of Text binds itself to the internal setText method within TTTAttributedLabel or not. The only difference this makes is if you want to set a NSAttributedString to use the .SetText( method, not .Text and not .AttiributedString as per the bug listed in the original source below.

 @bug Setting `attributedText` directly is not recommended, as it may cause a crash when attempting to access any links previously set. Instead, call `setText:`, passing an `NSAttributedString`.

beeradmoore commented:
After further inspection this will fail to build for device -_-

Commit:
Removed NSCheckingResult for now.
